pkg/specgen/generate/namespaces.go

```diff
@@ -42,6 +42,9 @@ func GetDefaultNamespaceMode(nsType string, cfg *config.Config, pod *libpod.Pod)
 			podMode = true
 		case nsType == "net" && pod.SharesNet():
 			podMode = true
+		case nsType == "net" && pod.NetworkMode() == "host":
+			toReturn.NSMode = specgen.Host
+			return toReturn, nil
```

Comment:
if the pod is sharing the netns and the mode is host, does the difference in the order here matter? I just tested `podman pod create --net=host --share=net` ad this still works

Reply:
No if the nets is not shared you still force host which is wrong:
```
$ bin/podman pod create --name test --share pid --net host
c7f39908615405fceb195e3db174e6b472f1415eb318d537b0e7e468ab4cf5f3
$ bin/podman run --pod test alpine ip a
<all host interfaces>
```
Since the netns is not shared the container should use the default not host.

Reply:
ahhh I see, I will make a follow up PR to fix this, I will also add checks for the other namespaces